File: 백준/Silver/10845. 큐/큐.cc
#include <bits/stdc++.h>
using namespace std;

int main(void) {
  int N;
  cin >> N;
  queue<int> Q;
  while (N--) {
    string cmd;
    cin >> cmd;
    if (cmd == "push") {
      int num;
      cin >> num;
      Q.push(num);
    } else if (cmd == "pop") {
      if (Q.empty()) {
        cout << -1 << '\n';
      } else {
        cout << Q.front() << '\n';
        Q.pop();
      }
    } else if (cmd == "size") {
      cout << Q.size() << '\n';
    } else if (cmd == "empty") {
      cout << Q.empty() << '\n';
    } else if (cmd == "front") {
      if (Q.empty()) {
        cout << -1 << '\n';
      } else {
        cout << Q.front() << '\n';
      }
    } else if (cmd == "back") {
      if (Q.empty()) {
        cout << -1 << '\n';
      } else {
        cout << Q.back() << '\n';
      }
    }
  }
}

File: 백준/Silver/10845. 큐/큐.py
import sys


def push(x):
    # queue.append(x)
    queue.insert(0, x)


def pop():
    if not queue:
        return -1
    else:
        # return queue.pop(0)
        return queue.pop()


def size():
    return len(queue)


def empty():
    if queue:
        return 0
    else:
        return 1


def front():
    if queue:
        return queue[-1]
    else:
        return -1


def back():
    if queue:
        return queue[0]
    else:
        return -1


N = int(sys.stdin.readline())
queue = []
for _ in range(N):
    command = list(sys.stdin.readline().split())
    if command[0] == "push":
        x = command[1]
        push(x)
    elif command[0] == "pop":
        print(pop())
    elif command[0] == "size":
        print(size())
    elif command[0] == "empty":
        print(empty())
    elif command[0] == "front":
        print(front())
    elif command[0] == "back":
        print(back())
